packages/driver/src/cy/commands/actions/type.ts

```diff
@@ -15,7 +14,8 @@ const debug = debugFn('cypress:driver:command:type')
 export default function (Commands, Cypress, cy, state, config) {
   const { keyboard } = cy.devices
 
-  function type (subject, chars, options = {}) {
+  // TODO: change the type of `any` to `Partial<Cypress.TypeOptions>`
+  function type (subject, chars, options: any = {}) {
```

Comment:
Cypress often extends then user-given `options` with some internal values like in line 24-34.

https://github.com/cypress-io/cypress/blob/bc87d6eae121cdbd31abd2c36ab1c788b36795b6/packages/driver/src/cy/commands/actions/type.ts#L24-L34

And TypeScript treats it as a type error. That's why it's commented. 

By the way, I tried to rename user-given `options` and internal options as `opts` in #6459. But it was rejected and refactored in the current style. Maybe we need to find a way to name our internal option objects to solve this problem.

Reply:
Can you add a small notes summarizing what you explained here? It makes sense to me but might not be obvious to others without doing a deeper drive into the code base on the work required here.

Reply:
The problem is that this comment exists on almost every command. So it's a bit hard to find the right place. 

And I'll remove all of them after removing `// @ts-nocheck` comments. (I believe the part 4 will be the final.)

Reply:
I estimated when removing `TODO`s will be done. And it might take 1-2 months in current speed. So, I decided to add a note above. 

I guess we need this somewhere in our code.